src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1501,40 +1692,72 @@ public DatabaseTopology ReadDatabaseTopology(BlittableJsonReaderObject rawDataba
             return JsonDeserializationCluster.DatabaseTopology(topology);
         }
 
-        public IEnumerable<(string Prefix, long Value)> ReadIdentities<T>(TransactionOperationContext<T> context, string databaseName, int start, long take)
-            where T : RavenTransaction
+        public IEnumerable<(string Prefix, long Value, long index)> GetIdentitiesFrom(TransactionOperationContext context, string dbName, long fromIndex, long take)
         {
-            var identities = context.Transaction.InnerTransaction.ReadTree(Identities);
+            CompareExchangeCommandBase.GetPrefixIndexSlices(context.Allocator, dbName, fromIndex, out var indexTuple);
 
-            var prefixString = UpdateValueForDatabaseCommand.GetStorageKey(databaseName, null);
-            using (Slice.From(context.Allocator, prefixString, out var prefix))
+            using (indexTuple.Scope)
             {
-                using (var it = identities.Iterate(prefetch: false))
+                var items = context.Transaction.InnerTransaction.OpenTable(IdentitiesSchema, Identities);
+                using (Slice.External(context.Allocator, indexTuple.Buffer, indexTuple.Buffer.Length, out var keySlice))
                 {
-                    it.SetRequiredPrefix(prefix);
+                    foreach (var tvr in items.SeekForwardFrom(IdentitiesSchema.Indexes[IdentitiesIndexSlice], keySlice, 0))
```

Comment:
Can read data from another db

Reply:
I have deleted this method and its CompareExchange reliative
Only left those who seek from given prefix